avm/res/cdn/profile/endpoint/main.bicep

```
@@ -0,0 +1,64 @@
+metadata name = 'CDN Profiles Endpoints'
+metadata description = 'This module deploys a CDN Profile Endpoint.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent CDN profile. Required if the template is used in a standalone deployment.')
+param profileName string
+
+@description('Required. Name of the endpoint under the profile which is unique globally.')
+param name string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Required. Endpoint properties (see https://learn.microsoft.com/en-us/azure/templates/microsoft.cdn/profiles/endpoints?pivots=deployment-language-bicep#endpointproperties for details).')
+param properties object
+
+@description('Optional. Endpoint tags.')
+param tags object?
+
+resource profile 'Microsoft.Cdn/profiles@2021-06-01' existing = {
+  name: profileName
+}
+
+resource endpoint 'microsoft.cdn/profiles/endpoints@2021-06-01' = {
+  parent: profile
+  name: name
+  location: location
+  properties: properties
+  tags: tags
+}
+
+module endpoint_origins 'origin/main.bicep' = [for origin in properties.origins: {
+  name: '${name}-origins-${origin.name}'
+  params: {
+    profileName: profile.name
+    endpointName: endpoint.name
+    name: origin.name
+    hostName: origin.properties.hostName
+    httpPort: origin.properties.?httpPort
+    httpsPort: origin.properties.?httpsPort
+    enabled: origin.properties.enabled
+    priority: origin.properties.?priority ?? -1
+    weight: origin.properties.?weight ?? -1
```

```suggestion
    priority: origin.properties.?priority
    weight: origin.properties.?weight
```

Should pick the default automatically